pkg/controller/update_cluster.go

```diff
@@ -308,6 +309,78 @@ func (c *Controller) addPodToPlacement(cluster *myspec.M3DBCluster, pod *corev1.
 	return nil
 }
 
+func (c *Controller) checkPodsForReplacement(
+	cluster *myspec.M3DBCluster,
+	pods []*corev1.Pod,
+	pl placement.Placement) (string, *corev1.Pod, error) {
+
+	var instancePodID myspec.PodIdentity
```

Comment:
Can we move this declaration to inside the `for` loop? I'm not sure what happens if on one iteration it has fields A and B, and the next object only deserialized field A, if B from the first loop would be left hanging

Reply:
yeah that makes more sense 👍